### PR TITLE
Fixes #20162 - Fix the modal for provisioning on index page.

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

**Comment by dLobatog** [comment]
[test]

**Comment by rabajaj0509** [comment]
Before:
modal

After:
modalfixed

File: app/views/discovered_hosts/_discovered_host_modal.html.erb [hunk 1]
```diff
@@ -18,13 +17,13 @@
...
-            <input type="submit" class="btn btn-default" value="<%= _('Quick Create') %>" name="quick_submit" />
-            <input type="submit" class="btn btn-primary" value="<%= _('Create Host') %>" />
-          </div>
-      <% end %>
+        <% end %>
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>
+        <input type="submit" class="btn btn-default" value="<%= _('Quick Create') %>" name="quick_submit" />
+        <input type="submit" class="btn btn-primary" value="<%= _('Customize') %>" />
```
Comment (dLobatog) [inline app/views/discovered_hosts/_discovered_host_modal.html.erb]: I don't really agree with that comment - https://groups.google.com/forum/?fromgroups#!topic/foreman-dev/q_6bxIsEKiU - `Create Host` is more accurate and expected IMO.

Comment (dLobatog) [reply app/views/discovered_hosts/_discovered_host_modal.html.erb]: cc @lzap @rohoover

**Review by dLobatog — CHANGES_REQUESTED** [review]
Thanks @rahulbajaj0509 

I'm not sure where the `well` is coming, but it makes a difference on the UI. Can you remove that css class so that it looks like the Patternfly implementation?

http://imgur.com/a/nNJmV

**Comment by rabajaj0509** [comment]
Thanks for the review @dLobatog, about the requested changes : 
1) 'well' comes from the 'foreman/foreman' project, if we change it there, then a lot of things will get changed. ( refer - https://github.com/theforeman/foreman/pull/4634 ) 
Therfore, my suggestion would be to keep this model as is.
2) I feel 'Create Host' would be appropriate 👍 , would still wait for lzap and Rohoover for thier input on this :) ( i made changes on the basis of evaluation by Roxanne 😛  )

**Comment by dLobatog** [comment]
@rohoover Can you take a look at the wording for "Create host"?

**Comment by Rohoover** [comment]
@dLobatog 

I think when we were doing the review it was brought up that the action wasn't creating a host but customizing because the following steps ask you to actually submit. I was trying to find the recording to verify but haven't located it yet.

**Comment by rabajaj0509** [comment]
@Rohoover i apologize, i had made these changes before going through the recording, i saw the recording and i was pretty sure that 'customize' is appropriate :) 

you can see the recording at : https://bluejeans.com/s/N0wwf

**Comment by lzap** [comment]
Sorry for the late reply - I vote for this wording (braces means blue color):

    Cancel - Customize host - [Create host]

Where Customize host is "Provision" and Create host is "Quick Create" in our old terms. I believe the quick path is more common.

**Comment by rabajaj0509** [comment]
@lzap i think that would be perfect :) Shall make the changes right away!

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
[test] - I think we have to rethink how permissions are added to locked roles in this plugin http://ci.theforeman.org/job/test_plugin_pull_request/3874/database=postgresql,label=fast,ruby=2.2/testReport/junit/(root)/CacheManagerTest/test_0001_new_filter_cache_can_be_created_regardless_of_locked_roles/

**Comment by dLobatog** [comment]
[test] - I think we have to rethink how permissions are added to locked roles in this plugin http://ci.theforeman.org/job/test_plugin_pull_request/3874/database=postgresql,label=fast,ruby=2.2/testReport/junit/(root)/CacheManagerTest/test_0001_new_filter_cache_can_be_created_regardless_of_locked_roles/

**Comment by dLobatog** [comment]
https://github.com/theforeman/foreman/pull/4657/files should unblock this PR

**Comment by rabajaj0509** [comment]
@dLobatog , thanks a lot. Executed the tests, all green 👍

**Comment by lzap** [comment]
[test]

**Review by lzap — CHANGES_REQUESTED** [review]
Buttons do not work for me, I click on them and nothing happens, except Cancel - that one do cancel tho.

**Comment by lzap** [comment]
Tests finally fixed [test]. Will re-test shortly.

**Comment by rabajaj0509** [comment]
Irrelevant tests are failing here, any idea why these tests are failing ?

**Comment by lzap** [comment]
Please rebase and push again, these were issues in core.

**Comment by rabajaj0509** [comment]
closing and re-opening for CI trigger.

**Comment by rabajaj0509** [comment]
[test]

**Comment by lzap** [comment]
Thanks.

**Pull request merged**